File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/HelloWorldCommand.java
package ro.razvan.java.dagger.atm;

import javax.inject.Inject;

public final class HelloWorldCommand implements NoArgsCommand {

    public static final String COMMAND_KEY_HELLO_WORLD = "hello";

    private final Outputter outputter;

    @Inject
    public HelloWorldCommand(Outputter outputter) {
        this.outputter = outputter;
    }

    @Override
    public String key() {
        return COMMAND_KEY_HELLO_WORLD;
    }

    @Override
    public Result handle() {
        outputter.output("world!");
        return Result.handled();
    }
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/HelloWorldCommand.kt
package ro.razvan.kotlin.dagger.atm

import ro.razvan.kotlin.dagger.atm.Command.Result
import javax.inject.Inject

const val COMMAND_KEY_HELLO_WORLD = "hello"

class HelloWorldCommand @Inject constructor(
    private val outputter: Outputter
): NoArgsCommand {

    override val key: String
        get() = COMMAND_KEY_HELLO_WORLD

    override fun handle(): Result {
        outputter("world!")
        return Result.handled()
    }
}